src/compiler/commandLineParser.ts

```diff
@@ -364,6 +364,7 @@ namespace ts {
 
         function parseStrings(args: string[]) {
             let i = 0;
+            function adjustFileName(arg: string): string { /*jshint -W110 */ return (arg[0] === "'" || arg[0] === '"') && arg[0] === arg[arg.length - 1] ? arg.substring(1, arg.length - 1) : arg; }

```

Comment:
also, i would only handle double quotes and not single. i do not think that is a common or expected support anyways.